sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -334,34 +339,51 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
   object Aggregation extends Strategy {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
       case PhysicalAggregation(
```

Comment:
Can we have another `case`  with `case ... if ...`? I think that's going to reduce the diff and makes it easier to manage, review, etc.

Reply:
Done. I changed to case matching.